cli/tests/test_describe_command.rs

```diff
@@ -781,6 +781,92 @@ fn test_describe_avoids_unc() {
     assert_eq!(edited_path, dunce::simplified(&edited_path));
 }
 
+#[test]
+fn test_describe_with_edit_and_message_args_opens_editor() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "repo"]);
+    let workspace_path = test_env.env_root().join("repo");
+
+    std::fs::write(workspace_path.join("file1"), "foo\n").unwrap();
+    std::fs::write(workspace_path.join("file2"), "bar\n").unwrap();
+    let edit_script = test_env.set_up_fake_editor();
+    std::fs::write(edit_script, ["dump editor"].join("\0")).unwrap();
+    let (stdout, stderr) = test_env.jj_cmd_ok(
+        &workspace_path,
+        &["describe", "-m", "message from command line", "--edit"],
+    );
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r"
+    Working copy now at: qpvuntsm 41069237 message from command line
+    Parent commit      : zzzzzzzz 00000000 (empty) (no description set)
+    ");
+    insta::assert_snapshot!(
+        std::fs::read_to_string(test_env.env_root().join("editor")).unwrap(), @r#"
+    message from command line
+
+    JJ: This commit contains the following changes:
+    JJ:     A file1
+    JJ:     A file2
+
+    JJ: Lines starting with "JJ:" (like this one) will be removed.
+    "#);
+}
+
+#[test]
+fn test_describe_change_with_existing_message_with_edit_and_message_args_opens_editor() {
```

Comment:
nit: I don't think we need both this test and the previous one. They're basically the same, but this one uses "original message" as initial description and the other test case uses "" (empty string) as initial description. They seem to take the same code paths too. But I don't feel strongly - feel free to leave both tests if you prefer.

Reply:
I've kept both tests.  When I first implemented this, I made a mistake and only covered one of the cases (the empty string case).  When I fixed that, I made it so that the code behaves the same in both cases, which means that theoretically you should only need one of the tests, but I figure leaving both helps in case someone else makes the same mistake I did.